app/vmctl/vm_native.go

```diff
@@ -303,3 +312,30 @@ func byteCountSI(b int64) string {
 	return fmt.Sprintf("%.1f %cB",
 		float64(b)/float64(div), "kMGTPE"[exp])
 }
+
+// buildMatchWithFilter adds additional filters from native.Filter.Match to discovered
+// metric names via /api/v1/series API.
+func buildMatchWithFilter(filter string, metricName string) (string, error) {
+	labels, err := promutils.NewLabelsFromString(filter)
+	if err != nil {
+		return "", err
+	}
+
+	var str strings.Builder
+
+	str.WriteString("{")
+	str.WriteString(nameLabel)
+	str.WriteString("=")
+	str.WriteString(strconv.Quote(metricName))
+	for _, label := range labels.Labels {
+		if label.Name == nameLabel {
+			continue
```

Comment:
It would be great leaving a comment on why the filter on the metric name is ignored here.